runtime/src/main/java/io/quarkiverse/logging/cloudwatch/LoggingCloudWatchHandler.java

```diff
@@ -97,54 +99,54 @@ public void close() throws SecurityException {
 
         @Override
         public void run() {
-            while (true) {
-                synchronized (eventBuffer) {
-                    events = new ArrayList<>(eventBuffer);
-                    eventBuffer.clear();
-                }
-
-                if (events.size() > 0) {
-                    PutLogEventsRequest request = new PutLogEventsRequest();
-                    request.setLogEvents(events);
-                    request.setLogGroupName(logGroupName);
-                    request.setLogStreamName(logStreamName);
+            synchronized (eventBuffer) {
+                events = new ArrayList<>(eventBuffer);
+                eventBuffer.clear();
+            }
+            boolean workingSequenceToken = false;
+            if (events.size() > 0) {
+                PutLogEventsRequest request = new PutLogEventsRequest();
+                request.setLogEvents(events);
+                request.setLogGroupName(logGroupName);
+                request.setLogStreamName(logStreamName);
+
+                while (!workingSequenceToken) {
```

Comment:
Calling an external service in a potentially infinite loop seems a bit risky. There should be a reasonnable limit (configurable or not) on how many iterations will be done.

If that limit is reached, it should throw an exception or log something at `SEVERE` level to let us know that we made too many attempts. We don't want to send too many alerts to Sentry though.